### DISCUSSION TITLE
[Feature] Update Postgres from version 14 (3 years old)

### DISCUSSION THREAD

**Discussion by Fmstrat** [immich-app/immich#14003] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

PG14 is getting dated and slow in many processes compared to 16. 17 was released last month, and pgvecto has 17 nightlies so far: https://hub.docker.com/r/tensorchord/pgvecto-rs/tags?name=pg17

I would like to request an update to the PG container, especially since it was 9 months since that image was built.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by alextran1502** [comment]
We will probably do an upgrade release before the stable release

**Reply by Fmstrat** [reply]
> We will probably do an upgrade release before the stable release

That's great to hear. It's one of those things that "works", but I'm constantly getting alerts about the age of the image. ;)

**Comment by Fmstrat** [comment]
Note: I am specifically discussing the docker-compose files in the repo that are used as a basis for running the system.

**Comment by bo0tzz** [comment]
Postgres 14 is still supported for two more years (https://www.postgresql.org/support/versioning/). You're free to run any newer version on your own setup, but an upgrade is not straightforward and so (like Alex mentions) we're not planning to do that until before the stable release.

**Reply by Fmstrat** [reply]
This makes sense, the main rationale for me is since the built image being used is so old (9 months) I get security alerts for it's age repeatedly.

**Comment by Fmstrat** [comment]
And geeeeeez. You guys are on it! That was like, 1 minute.
